pkgs/games/maptool/default.nix

```diff
@@ -99,15 +97,20 @@ stdenvNoCC.mkDerivation {
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin
-    makeWrapper ${jre}/bin/java $out/bin/${binName} \
-      "''${gappsWrapperArgs[@]}" \
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ ffmpeg ]} \
-      --add-flags '${lib.concatStringsSep " " jvmArgs} net.rptools.maptool.client.LaunchInstructions'
+    dest=$out/${appClasspath}
+    install -dm755 "$dest"
+    bsdtar -xf "$src" -C "$dest" --strip-components 4 opt/maptool/lib/app/{'*.jar',readme}
 
     dest=$out/share/icons/hicolor/256x256/apps
-    mkdir -p "$dest"
-    ln -s ${icon} "$dest/${rdnsName}.png"
+    install -dm755 "$dest"
+    bsdtar -xOf "$src" opt/maptool/lib/MapTool.png > "$dest"/${rdnsName}.png
+
+    dest=$out/bin
+    install -dm755 "$dest"
+    makeWrapper ${jre}/bin/java "$dest"/${binName} \
+      "''${gappsWrapperArgs[@]}" \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ ffmpeg ]} \
```

Comment:
Is there an alternative to LD_LIBRARY_PATH?

Reply:
This package uses hundreds of pre-compiled JARs, many of which contain binary libraries. The only alternative I can think of would be unpacking those JARs, patchelf-ing each library, and then reassembling the JARs, and I really think using `LD_LIBRARY_PATH` is less brittle than that.